edx/frontend-app-payment
# chore: [REV-1551] clean up dependencies for Renovate

## Description
This repository's build currently fails. To fix:

* Remediate or whitelist `npm audit` findings.
    * Build `package-lock.json` without upstream vulns via `npx npm-force-resolutions`.
    * Whitelist moderate findings unpatchable by `npm-force-resolutions`.
    * Whitelist high finding [CVE-2021-23424](https://github.com/advisories/GHSA-whgm-jr23-g3j9) pending patch.
* Update dependencies with breaking changes in preparation for Renovate.
* Fix broken tests that made its way into master.

Also, perform other repository clean up tasks:

* Add to PR template reminders for:
    * PCI compliance.
    * expedient deployment.
* Move audit check order to last in CI to make test failures more visible.
* Have Makefile `precommit` act the same as the package.json precommit.
* Get precommit checks working again.
* Update README.md.

## Supporting information

* JIRA ticket: [REV-1551](https://openedx.atlassian.net/browse/REV-1551)
* Research Notes: [Confluence](https://openedx.atlassian.net/wiki/spaces/~931919476/pages/3229155368/REV-2388#Dependency-Issues)

## Testing instructions

On devstack, no webpack errors and able to complete purchase via web UI.

Check on stage prior to deployment:
* [ ] Test New Relic browser agent and traces work.
* [x] Check if `nasm` needs to be installed on stage.
* [x] Check if `nasm` or `ZLIB_1.2.9` is available on production.
* [ ] Manually run e2e tests.
* [ ] Check the following CSS classes properly render:
    * [ ] `a.pgn__hyperlink.default-link.standaline-link`
    * [ ] `svg.fa-null.fa-rotate-null.fa-pull-null`

After deploy to Prod, besides usual monitoring:
* [ ] Perform test purchase via Cybersource
* [ ] Perform test purchase via PayPal

## Other Information

`npm-force-resolutions` config:

```
"resolutions": {
    "immer": ">=9.0.6",
    "trim-newlines": ">=3.0.1"
}
```

## Checklist
- [X] Consider PCI compliance impact and whether this PR changes how credit card information is handled.
- [X] Intend to [release and monitor](https://openedx.atlassian.net/wiki/spaces/RS/pages/1835106870/How+to+contribute+to+our+repositories) this PR promptly after merge.
